src/config/wmodules-config.c

```diff
@@ -219,6 +219,63 @@ int Read_SCA(const OS_XML *xml, xml_node *node, void *d1)
     return 0;
 }
 
+int Read_GCP(const OS_XML *xml, xml_node *node, void *d1) {
+    wmodule **wmodules = (wmodule**)d1;
+    wmodule *cur_wmodule;
+    xml_node **children = NULL;
+    wmodule *cur_wmodule_exists;
+
+    // Allocate memory
+    if ((cur_wmodule = *wmodules)) {
+        cur_wmodule_exists = *wmodules;
+
+        while (cur_wmodule_exists) {
+            if(cur_wmodule_exists->tag) {
+                if(strcmp(cur_wmodule_exists->tag,node->element) == 0) {
+                    cur_wmodule = cur_wmodule_exists;
+                    break;
+                }
+            }
+
+            if (cur_wmodule_exists->next == NULL) {
+                while (cur_wmodule->next) {
+                    cur_wmodule = cur_wmodule->next;
+                }
```

Comment:
After running this loop, cur_wmodule will be equal to cur_wmodule_exists, so this could be simplified a little more.